public/scripts/project/metadata.js

```diff
@@ -127,17 +151,50 @@ define(function(require) {
       url: url
     });
     request.done(function(data) {
-      setMetadata(root, data, callback);
+      callback(data);
     });
     request.fail(function(jqXHR, status, err) {
       callback(err);
     });
   }
 
+  function loadAnonymousMetadata(config, callback) {
+    callback = lockSafeCallback(callback);
+
+    lock(function() {
+      fs.getxattr(root, PROJECT_META_KEY, function(err, value) {
+        if(err) {
+          if(err.code !== 'ENOATTR') {
+            return callback(err);
+          }
+          return setAnonymousMetadata(config, callback);
+        }
+
+        callback();
+      });
+    });
+  }
+
+  function loadMetadata(config, callback) {
+    if (config.user) {
+      return fetchMetadata(config, function(err, data) {
+        setMetadata({
+          root: config.root,
+          data: data,
+          user: config.user,
+          title: config.title
+        }, callback);
+      });
+    }
+
+    loadAnonymousMetadata(config, callback);
+  }
+
   return {
     loadMetadata: loadMetadata,
     getFileID: getFileID,
     setFileID: setFileID,
+    getTitle: getTitle,
```

Comment:
Do you not need `setTitle` for the case that the user updates the title on the project?  How do we handle that?  We should keep all that state in here if possible.